pkg/apis/core/validation/shoot.go

```diff
@@ -2382,3 +2477,19 @@ func isShootReadyForRotationStart(lastOperation *core.LastOperation) bool {
 	}
 	return lastOperation.Type == core.LastOperationTypeReconcile
 }
+
+func getResourcesForEncryption(apiServerConfig *core.KubeAPIServerConfig) []string {
+	resources := sets.New[string]()
+
+	if apiServerConfig != nil && apiServerConfig.EncryptionConfig != nil {
+		for _, res := range apiServerConfig.EncryptionConfig.Resources {
+			resources.Insert(res)
+		}
+	}
+
+	if resources.Len() > 0 {
+		sets.List(resources)
```

Comment:
```suggestion
		return sets.List(resources)
```
?

Reply:
Apparently `reflect.DeepEqual` returns false if one is empty slice and one is nil, Will use `apiequality.Semantic.DeepEqual` instead.